x-pack/plugin/snapshot-repo-test-kit/src/main/java/org/elasticsearch/repositories/blobstore/testkit/RepositoryAnalyzeAction.java

```diff
@@ -542,7 +542,7 @@ private BlobContainer getBlobContainer() {
 
         private void onWorkerCompletion() {
             if (workerCountdown.countDown()) {
-                transportService.getThreadPool().executor(ThreadPool.Names.SNAPSHOT).execute(ActionRunnable.wrap(listener, l -> {
+                transportService.getThreadPool().executor(ThreadPool.Names.SAME).execute(ActionRunnable.wrap(listener, l -> {
```

Comment:
You don't really need `SAME` if you don't want to fork, you can just run the action outright. But what thread is this even running on, we can't run any IO on a transport thread if that's where this is called from?

Reply:
Ah I see, let me investigate and tell you what thread it is. But in any case @original-brownbear , do you think we could use another thread pool then to send the IO response?

Reply:
I suppose in theory we could always use some other pool like `GENERIC` here but I wonder if that's a wise solution. Maybe it'd be better to just adjust the settings of the node to a larger snapshot pool size and/or adjust the test case to a smaller scenario that doesn't use up the full pool, instead of adjusting prod code to work around a test scenario that is impossible in the real world?

Reply:
@original-brownbear , decreasing the test is also possible to go around this! I had not thought of that. Lets do that indeed. I will reshape the PR and then ask for your review once more.